src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -594,6 +624,61 @@ def post_process_containerapp(self, r):
                 linker_client.linker.begin_create_or_update(resource_uri=r["id"],
                                                             parameters=item["parameters"],
                                                             linker_name=item["linker_name"]).result()
+
+        if self.get_argument_repo():
+            app = self.set_up_create_containerapp_if_source_or_repo()
+            r = self.set_up_create_containerapp_repo(app=app, r=r, env=app.env, env_rg=app.resource_group.name)
+        return r
+
+    def set_up_create_containerapp_if_source_or_repo(self):
+        from ._up_utils import (ContainerApp, ResourceGroup, ContainerAppEnvironment, _reformat_image)
+
+        # Parse resource group name and managed env name
+        env_id = self.containerapp_def["properties"]['environmentId']
+        parsed_managed_env = parse_resource_id(env_id)
+        env_name = parsed_managed_env['name']
+        env_rg = parsed_managed_env['resource_group']
+
+        # Parse location
+        env_info = self.get_environment_client().show(cmd=self.cmd, resource_group_name=env_rg, name=env_name)
+        location = env_info['location']
+
+        # Set image to None if it was previously set to the default image (case where image was not provided by the user) else reformat it
+        image = None if self.get_argument_image().__eq__(HELLO_WORLD_IMAGE) else _reformat_image(self.get_argument_source(), self.get_argument_repo(), self.get_argument_image())
```

Comment:
Do you mind explaining what the image reformat does? It seems split the image by the `/` character, take the last item, and remove the `:` character. I'm curious why it is doing this?

Reply:
This `_reformat_image` method essentially just renames the repo to original repo name + tag
If the user provided the `image` arg with `source` or `repo`, the builder generates a new image and is named as registry-server-name/re-formated image:datetimetag (for source) or registry-server-name/re-formated image:github sha (for repo). Apart from re-naming the image (associated with the container app that will be created) this arg does not have value when used with source or repo. We did it this way to be consistent with what the `containerapp up` command is doing.

Example - `az containerapp create -g rg -n capp-name --source . --image random.azurecr.io/randomrepo:latest --registry-server random.azurecr.io --registry-user random --registry-pass random_password` will create a container app `capp-name` and the image associated with this container app will be named `random.azurecr.io/randomrepolatest:datetime`.

This image was generated by either the dockerfile (if the source repo had a dockerfile) or build packs (if the project was a dotnet 7 or a node 18 project) or acr build task if it did not satisfy the first two conditions.

One other thing to keep in mind is that the registry-server, registry username and password are required arguments if user used source/repo